autotx/helper_agents/clarifier.py

```diff
@@ -0,0 +1,86 @@
+        """ if not not_interactive else "")
+
+    missing_2 = dedent("""
+        Analyze the user's initial goal if there is missing information, and ask the user for it. 
+        E.g. "Buy ETH" -> "How much ETH do you want to buy and with what token?"
+        """ if not not_interactive else "" )
+
+    missing_3 = dedent("""
+        - Call the request_user_input tool if more information is needed to define the goal.
+        """ if not not_interactive else "")
+    
+    clarifier_agent = AssistantAgent(
+        name="clarifier",
+        system_message=dedent(
+            f"""
+            Clarifier is an assistant that can analyze a user's goal at the start of the conversation and determine if it is within the domain of the agents.
+            The user will provide the goal, and the agents are meant to help prepare one or more necessary transactions to accomplish the goal.
+            When dealing with Ethereum transactions, assume the following:
+                - The user's address: {smart_account_addr}
+                - The network to interact with: {network_name} 
+                - When the user want to exeucute transactions he means to prepare the transactions.
+
+            You must analyze the goal to be executed by the agents.
+            If the goal is invalid or outside the scope of the agents, you MUST call the goal_outside_domain tool.
+            Only call goal_outside_domain if the goal is outside of the domain of the agents.
+            {
+                missing_1
+            }
+            DO NOT make any assumptions about the user's intent or context and ALWAYS take into account the available tools and their descriptions.
+            
+            The available agents and tools:
+            {agents_information}
+            {
+                missing_2
+            }
+
+            The only things the clarifier should do are:
+            {
+                missing_3
+            }
+            - Call the goal_outside_domain tool if the goal is outside the domain of the agents.
+            - Nothing
+            Perform these actions ONLY in the BEGINNING of the conversation.
+            """
+        ),
+        llm_config=get_llm_config(),
+        human_input_mode="NEVER",
+        code_execution_config=False,
+    )
+
+    if not not_interactive:
+        def request_user_input(
+            message: Annotated[str, "The message to ask the user for input"],
+        ) -> str:
+            return input(f"Question: {message}\nYour reply: ")
+
+        clarifier_agent.register_for_llm(name="request_user_input", description="Request user input")(request_user_input)
+        user_proxy.register_for_execution(name="request_user_input")(request_user_input)
+
+    def goal_outside_domain(
+        message: Annotated[str, "The message return to the user about why the goal is outside of the supported domain"],
+    ) -> str:
+        cprint(f"Goal not supported: {message}", "red")
+        return "TERMINATE"
```

Comment:
previously, if the user entered a prompt that wasn't supported, we would give the option to input a new goal - i think we should keep the same behavior here

Reply:
done